### PR TITLE
adding trending_now sample code notebooks

### PR DESCRIPTION
**Pull request by enabov**
*Issue #, if available:*

*Description of changes:*
Adding sample notebook code for new Trending-Now recipe

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: next_steps/core_use_cases/trending_now/trending_now_interactions_data_prep.ipynb
Comment (james-jory) [inline next_steps/core_use_cases/trending_now/trending_now_interactions_data_prep.ipynb:1]: What is the purpose of this notebook? Seems to overlap with the trending_now_example notebook.

File: next_steps/core_use_cases/trending_now/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Amazon Personalize Trending-Now Recipe
+---
+
+The sample notebook (trending_now_example.ipynb) will walk you through an example of using Trending Now recipe in [Amazon Personalize]
```
[the remark this answers is not in the document]
Comment (enabov) [reply next_steps/core_use_cases/trending_now/README.md]: done, added the hyperlink for Amazon Personalize PDP in latest commit

File: next_steps/core_use_cases/trending_now/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+User interests can change based on a variety of factors, such as external events or the interests of other users. It is critical for websites and apps to tailor their recommendations to these changing interests to improve user engagement. With Trending-Now, you can surface items from your catalogue that are rising in popularity faster with higher velocity than other items, such as trending news, popular social content or newly released movies. Amazon Personalize looks for items that are rising in popularity at a faster rate than other catalogue items to help users discover items that are engaging their peers. 
+
+Amazon Personalize also allows customers to define the time periods over which trends are calculated depending on their unique business context, with options for every 30 mins, 1 hour, 3 hours or 1 day, based on the most recent interactions data from users. This notebook will demonstrate how the new recipe aws-trending-now (or aws-vod-trending-now for recommenders) can help recommend the top trending items from the interactions dataset.
```
[the remark this answers is not in the document]
Comment (enabov) [reply next_steps/core_use_cases/trending_now/README.md:8]: done, added the hyperlink for Trending-Now in latest commit

File: next_steps/core_use_cases/trending_now/trending_now_example.ipynb [hunk 3]
```diff
@@ -0,0 +1,1574 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "00a89f03",
+   "metadata": {},
+   "source": [
+    "# Amazon Personalize introduces - Trending Now\n",
+    "\n",
+    "This notebook will walk you through an example of using Trending Now recipe in [Amazon Personalize]\n",
```
[the remark this answers is not in the document]
Comment (enabov) [reply next_steps/core_use_cases/trending_now/trending_now_example.ipynb]: done, added the hyperlink for Amazon Personalize PDP in latest commit

File: next_steps/core_use_cases/trending_now/trending_now_example.ipynb [hunk 4]
```diff
@@ -0,0 +1,1574 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "58dd1b55",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "items_df.describe()"
+   ]
+  },
```
[the remark this answers is not in the document]
Comment (enabov) [reply next_steps/core_use_cases/trending_now/trending_now_example.ipynb:290]: Added a comment that we are creating Items dataset for the recommendations readability

File: next_steps/core_use_cases/trending_now/trending_now_example.ipynb [hunk 5]
```diff
@@ -0,0 +1,1574 @@
...
+    "personalize.describe_solution_version(solutionVersionArn=solution_version_arn)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "904d90d9",
+   "metadata": {},
+   "source": [
+    "## Campaigns\n",
+    "With solution and solution version created, we will now create a campaign. Please note, with trending now recipe, we can use both filter and promotions while calling getRecommendations over an active campaign (or recommender).\n",
```
[the remark this answers is not in the document]
Comment (enabov) [reply next_steps/core_use_cases/trending_now/trending_now_example.ipynb]: done, added hyperlinks for filters and promotions

File: next_steps/core_use_cases/trending_now/trending_now_example.ipynb [hunk 6]
```diff
@@ -0,0 +1,1574 @@
...
+    "Campaigns"
```
[the remark this answers is not in the document]
Comment (enabov) [reply next_steps/core_use_cases/trending_now/trending_now_example.ipynb:843]: updated the sentence to close it

File: next_steps/core_use_cases/trending_now/trending_now_example.ipynb [hunk 7]
```diff
@@ -0,0 +1,1574 @@
...
+    "recommendations.head(10)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cfccd2ec",
+   "metadata": {},
+   "source": [
+    "## EventTracker to ingest Put Events to trigger auto-updates\n",
+    "To use this recipe effectively and to catch trendiness in real time - one should use putEvents API to continuously ingest new events into the system. Trending now model will incorporate these new events with automatic model updates (at no cost to end customer) to show latest trending now items.\n",
```
[the remark this answers is not in the document]
Comment (enabov) [reply next_steps/core_use_cases/trending_now/trending_now_example.ipynb]: updated the statement to specify it is based on Trend discovery frequency configured

File: next_steps/core_use_cases/trending_now/trending_now_example.ipynb [hunk 8]
```diff
@@ -0,0 +1,1574 @@
...
+   "source": [
+    "for i in range(0,25):\n",
+    "    for user_id in user_rand:\n",
+    "        for item_id in items_rand:\n",
+    "            time.sleep(1)\n",
+    "            time_epoch = int(time.time())\n",
+    "            response = personalize_events.put_events(\n",
+    "                trackingId=tracking_id,\n",
+    "                userId=str(user_id),\n",
+    "                sessionId=str(user_id),\n",
```
[the remark this answers is not in the document]
Comment (enabov) [reply next_steps/core_use_cases/trending_now/trending_now_example.ipynb]: updated code to use UUID as the sessionId

File: next_steps/core_use_cases/trending_now/trending_now_example.ipynb [hunk 9]
```diff
@@ -0,0 +1,1574 @@
...
+    "                eventList=[\n",
+    "                    {\n",
+    "                        'eventType': str(user_id),\n",
```
[the remark this answers is not in the document]
Comment (enabov) [reply next_steps/core_use_cases/trending_now/trending_now_example.ipynb]: EVENT_TYPE is not required for Trending-Now recipe, so removed eventType from the put_events api

File: next_steps/core_use_cases/trending_now/trending_now_example.ipynb [hunk 10]
```diff
@@ -0,0 +1,1574 @@
...
+    "\n",
+    "        # 5. Delete datasets and their schemas\n",
+    "        _delete_datasets_and_schemas(dataset_group_arn)\n",
+    "\n",
+    "        # 6. Delete dataset group\n",
+    "        _delete_dataset_group(dataset_group_arn)\n",
+    "\n",
+    "        logger.info(f'Dataset group {dataset_group_arn} fully deleted')"
+   ]
+  },
```
[the remark this answers is not in the document]
Comment (enabov) [reply next_steps/core_use_cases/trending_now/trending_now_example.ipynb:1466]: Actually the idea was to provide a single notebook for the blog reader to implement the solution. So I had combined the clean up also as part of this notebook.

File: next_steps/core_use_cases/trending_now/trending_now_example.ipynb (hunk 9 above)
[the remark this answers is not in the document]
Comment (james-jory) [reply next_steps/core_use_cases/trending_now/trending_now_example.ipynb]: `eventType` is required in the [Event payload](https://docs.aws.amazon.com/personalize/latest/dg/API_UBS_Event.html) and didn't think PutEvents was smart enough to check the schema to see if `EVENT_TYPE` was specified. If you tested this and it works, will chalk it up to TIL!

File: next_steps/core_use_cases/trending_now/trending_now_example.ipynb (hunk 4 above)
[the remark this answers is not in the document]
Comment (james-jory) [reply next_steps/core_use_cases/trending_now/trending_now_example.ipynb:290]: But the notebook is still creating what it calls an "items dataset" and renames the columns to conform to an items dataset but it never does anything with it as an actual Personalize items dataset. It's just used as a dataframe for lookups of movie titles to test inference. Obviously it works as-is but it unnecessarily confuses the issue. Will let it go.

File: next_steps/core_use_cases/trending_now/trending_now_example.ipynb [hunk 11]
```diff
@@ -0,0 +1,1574 @@
...
+    "    time.sleep(60)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "08651e6f",
+   "metadata": {},
+   "source": [
+    "## Training the Trending-Now model - Solution & Solution Version\n",
+    "With the interactions and items datasets imported into dataset group, we will next create solutions and solution versions using the trending-now recipe. First, let's initialize the trending-now-recipe for creating the custom solution. Please note the new featureTransformationParameter that we are setting for this recipe while creating the solution. Its value can be selected from the set of [30 minutes, 1 hour, 3 hours, 1 day].\n",
```
[the remark this answers is not in the document]
Comment (james-jory) [reply next_steps/core_use_cases/trending_now/trending_now_example.ipynb]: Nitpick: it's still referring to "datasets", "solutions", and "solution versions". Should be singular (i.e., "dataset", "solution", and "solution version").

File: next_steps/core_use_cases/trending_now/trending_now_incremental_data_prep.ipynb
[the remark this answers is not in the document]
Comment (james-jory) [reply next_steps/core_use_cases/trending_now/trending_now_incremental_data_prep.ipynb:1]: So this only applies if the user is using the console and not the end-to-end notebook approach? Do we have a precedence for notebooks covering full notebook and hybrid console & notebook? This was definitely not clear to me upon first review. There's nothing in the README that explains what this notebook is for and the title and first paragraph in the notebook don't help clarify this either.

File: next_steps/core_use_cases/trending_now/trending_now_interactions_data_prep.ipynb
Comment (james-jory) [reply next_steps/core_use_cases/trending_now/trending_now_interactions_data_prep.ipynb:1]: But that's what the notebook above is supposed to be doing? How is this notebook different? I'm still confused by this notebook is necessary.

**Review by james-jory — APPROVED** [review]
Looks great! Thanks for the patience on the review. ;-)

**Pull request merged**